test/core/promise/exec_ctx_wakeup_scheduler_test.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/promise/exec_ctx_wakeup_scheduler.h"
+
+#include <gtest/gtest.h>
+
+#include "src/core/lib/promise/activity.h"
+
+namespace grpc_core {
+
+TEST(ExecCtxWakeupSchedulerTest, Works) {
+  int state = 0;
+  bool done = false;
+  auto activity = MakeActivity(
+      [&state]() mutable -> Poll<absl::Status> {
+        ++state;
+        switch (state) {
+          case 1:
+            return Pending();
```

Comment:
Do I understand correctly that if this activity/lambda called `ExecCtx::Run` from within, the normal ExecCtx mechanisms would schedule and execute the closure?

Reply:
Correct.

Reply:
thanks for confirming.